File: Actividad2MODULO3DanielRomero/app/src/main/java/com/example/actividad2modulo3danielromero/MainActivity.java
package com.example.actividad2modulo3danielromero;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.EditText;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    TextView total;
    Button sum, mult, div, resta;
    EditText n1, n2;
    float n1_float, n2_float, total_float = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        sum = (Button) findViewById(R.id.sum);
        mult = (Button) findViewById(R.id.mult);
        resta = (Button) findViewById(R.id.resta);
        div = (Button) findViewById(R.id.div);

        n1 = (EditText) findViewById(R.id.n1);
        n2 = (EditText) findViewById(R.id.n2);
        total = (TextView) findViewById(R.id.total);


        System.out.println("----------------------------------------------------");


        sum.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                System.out.println(n1.getText());
                if (!n1.getText().equals("")) {
                    n1_float = Float.parseFloat(n1.getText()+"");
                } else {
                    n1_float = 0;
                }
                if (!n2.getText().equals("")) {
                    n2_float = Float.parseFloat(n2.getText()+"");
                } else {
                    n2_float = 0;
                }

                total_float = n1_float + n2_float;

                Log.d("MainActivity", "Calculadora - Operacion Suma");

                total.setText(total_float+"");

            }
        });

        resta.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                if (!n1.getText().equals("")) {
                    n1_float = Float.parseFloat(n1.getText()+"");
                } else {
                    n1_float = 0;
                }
                if (!n2.getText().equals("")) {
                    n2_float = Float.parseFloat(n2.getText()+"");
                } else {
                    n2_float = 0;
                }

                total_float = n1_float - n2_float;
                Log.d("MainActivity", "Calculadora - Operacion Resta");

                total.setText(total_float+"");

            }
        });

        mult.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                if (!n1.getText().equals("")) {
                    n1_float = Float.parseFloat(n1.getText()+"");
                } else {
                    n1_float = 0;
                }
                if (!n2.getText().equals("")) {
                    n2_float = Float.parseFloat(n2.getText()+"");
                } else {
                    n2_float = 0;
                }

                total_float = n1_float * n2_float;
                Log.d("MainActivity", "Calculadora - Operacion Multiplicación");

                total.setText(total_float+"");

            }
        });

        div.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View v) {
                if (!n1.getText().equals("")) {
                    n1_float = Float.parseFloat(n1.getText()+"");
                } else {
                    n1_float = 0;
                }
                if (!n2.getText().equals("")) {
                    n2_float = Float.parseFloat(n2.getText()+"");
                } else {
                    n2_float = 1;
                }

                total_float = n1_float/n2_float;
                Log.d("MainActivity", "Calculadora - Operacion División");

                total.setText(total_float+"");

            }
        });
    }
}

File: C2_MODULO4_DANIELROMERO/app/src/main/java/com/root/c2_modulo4_danielromero/MainActivity.kt
package com.root.c2_modulo4_danielromero

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    var total: TextView? = null
    var sumBtn: Button? = null
    var restaBtn: Button? = null
    var multBtn: Button? = null
    var divBtn: Button? = null
    var cifra01: EditText? = null
    var cifra02: EditText? = null
    var c1Float = 0.0
    var c2Float = 0.0
    var totalFloat = 0.0



    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        total = findViewById<View>(R.id.total_out) as TextView
        sumBtn = findViewById<View>(R.id.sum_btn) as Button
        restaBtn = findViewById<View>(R.id.resta_btn) as Button
        multBtn = findViewById<View>(R.id.mult_btn) as Button
        divBtn = findViewById<View>(R.id.div_btn) as Button
        cifra01 = findViewById<View>(R.id.c1_in) as EditText
        cifra02 = findViewById<View>(R.id.c2_in) as EditText

        sumBtn!!.setOnClickListener {
            if ( !cifra01!!.text.toString().trim { it <= ' '}.isEmpty()) {
                c1Float = (cifra01!!.text.toString() + "").toDouble()
            } else {
                c1Float = 0.0
                cifra01!!.setText("0")
            }
            if ( !cifra02!!.text.toString().trim { it <= ' '}.isEmpty()) {
                c2Float = (cifra02!!.text.toString() + "").toDouble()
            } else {
                c2Float = 0.0
                cifra02!!.setText("0")
            }

            totalFloat = c1Float + c2Float
            total!!.text = totalFloat.toString() + ""

        }

        restaBtn!!.setOnClickListener {
            if ( !cifra01!!.text.toString().trim { it <= ' '}.isEmpty()) {
                c1Float = (cifra01!!.text.toString() + "").toDouble()
            } else {
                c1Float = 0.0
                cifra01!!.setText("0")
            }
            if ( !cifra02!!.text.toString().trim { it <= ' '}.isEmpty()) {
                c2Float = (cifra02!!.text.toString() + "").toDouble()
            } else {
                c2Float = 0.0
                cifra02!!.setText("0")
            }

            totalFloat = c1Float - c2Float
            total!!.text = totalFloat.toString() + ""

        }

        multBtn!!.setOnClickListener {
            if ( !cifra01!!.text.toString().trim { it <= ' '}.isEmpty()) {
                c1Float = (cifra01!!.text.toString() + "").toDouble()
            } else {
                c1Float = 0.0
                cifra01!!.setText("0")
            }
            if ( !cifra02!!.text.toString().trim { it <= ' '}.isEmpty()) {
                c2Float = (cifra02!!.text.toString() + "").toDouble()
            } else {
                c2Float = 0.0
                cifra02!!.setText("0")
            }

            totalFloat = c1Float * c2Float
            total!!.text = totalFloat.toString() + ""

        }

        divBtn!!.setOnClickListener {
            if ( !cifra01!!.text.toString().trim { it <= ' '}.isEmpty()) {
                c1Float = (cifra01!!.text.toString() + "").toDouble()
            } else {
                c1Float = 1.0
                cifra01!!.setText("1")
            }
            if ( !cifra02!!.text.toString().trim { it <= ' '}.isEmpty()) {
                c2Float = (cifra02!!.text.toString() + "").toDouble()
            } else {
                c2Float = 1.0
                cifra02!!.setText("1")
            }

            totalFloat = c1Float / c2Float
            total!!.text = totalFloat.toString() + ""

        }
    }

//    fun sumar(view: View?){
//        val result = cifra_01!!.text.toString().toInt() + cifra_02!!.text.toString().toInt()
//        resultado!!.text = result.toString() + ""
//        Log.d("operacionSuma", "Boton suma apretado")
//    }
}